### PR TITLE
Add storage interface and database storage class

### PR DESCRIPTION
**Pull request by andrewnester**

### CODE REVIEW

File: app/Storages/DatabaseStorage.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    private $modelClass;
+
+    public function __construct($modelClass)
+    {
+        $this->modelClass = $modelClass;
+    }
+
+    public function all($columns = ['*'])
+    {
+        return call_user_func_array($this->modelClass.'::all', [$columns]);
```
Comment (andrewnester) [inline app/Storages/DatabaseStorage.php]: need to use QueryBuilder here

File: app/Storages/DatabaseStorage.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    {
+        return $obj->delete();
+    }
+
+    public function getQueryBuilder()
+    {
+        $model = new $this->modelClass;
+        return $model->newQuery();
+    }
+}
```
Comment (andrewnester) [inline app/Storages/DatabaseStorage.php]: add new line

File: app/Storages/DatabaseStorage.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    public function all($columns = ['*'])
+    {
+        return call_user_func_array($this->modelClass.'::all', [$columns]);
+    }
+
+    public function one($id, $columns = ['*'])
+    {
+        return call_user_func_array($this->modelClass.'::find', [$id, $columns]);
```
Comment (andrewnester) [inline app/Storages/DatabaseStorage.php]: need to use QueryBuilder here

File: app/Storages/DatabaseStorage.php [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+ *
+ * @package App\Storages
+ */
+class DatabaseStorage implements StorageContract
+{
+    private $modelClass;
+
+    public function __construct($modelClass)
+    {
+        $this->modelClass = $modelClass;
```
Comment (andrewnester) [inline app/Storages/DatabaseStorage.php]: need to pass instance not class name

File: app/Providers/AppServiceProvider.php [hunk 5]
```diff
@@ -28,11 +27,11 @@ public function boot()
     public function register()
     {
         $this->app->singleton('App\Contracts\CategoriesServiceContract', function () {
-            return new CategoriesService(new Category());
+            return new CategoriesService(new DatabaseStorage('\App\Category'));
```
Comment (andrewnester) [inline app/Providers/AppServiceProvider.php]: need to pass model instance, not its name

File: app/Providers/AppServiceProvider.php [hunk 6]
```diff
@@ -28,11 +27,11 @@ public function boot()
...
         });
 
         $this->app->singleton('App\Contracts\ItemsServiceContract', function() {
-            return new ItemsService(new Item(), $this->app['App\Contracts\CategoriesServiceContract']);
+            return new ItemsService(new DatabaseStorage('\App\Item'), $this->app['App\Contracts\CategoriesServiceContract']);
```
Comment (andrewnester) [inline app/Providers/AppServiceProvider.php]: need to pass model instance, not its name

File: storage/framework/sessions/.gitignore [hunk 7]
```diff
@@ -0,0 +1,2 @@
+*
+!.gitignore
```
Comment (nester-fish) [inline storage/framework/sessions/.gitignore]: add empty line

File: storage/framework/views/.gitignore [hunk 8]
```diff
@@ -0,0 +1,2 @@
+*
+!.gitignore
```
Comment (nester-fish) [inline storage/framework/views/.gitignore]: add empty line

**Comment by nester-fish** [comment]
Should be redone

**Pull request closed without merging**